### PR TITLE
0.4.1 release blog post.

### PR DESCRIPTION
**Pull request by SamBarker**

### CODE REVIEW

File: _posts/2024-01-12-release-0_4_1.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+The diagram below shows how the system works at a high level.
+
+![image](https://github.com/kroxylicious/kroxylicious.github.io/assets/18440250/09aeb1f4-f420-4d89-b9f7-394473dddc05){:width="100%"}
+
+If this use-case interests you, here's a [short demo](https://asciinema.org/a/627637) that shows the feature in action. To try it out yourself there's some [interim instructions](https://github.com/kroxylicious/kroxylicious/blob/1e58fc703e45c7fdb0990adb583cf4aa857047cd/kubernetes-examples/envelope-encryption/README.md) to use the feature in development.  There will be more information landing on the website soon, but meanwhile, if you have questions or comments please head over to the [Slack channel](https://kroxylicious.slack.com/).
+
+{% include asciicinema.html id="627637" %}
+
+### Binary artefacts
```
Comment (tombentley) [inline _posts/2024-01-12-release-0_4_1.md:33]: I think we should stick to the American spelling (artifact). It's what people are used to seeing (e.g. in `pom.xml`).

File: _posts/2024-01-12-release-0_4_1.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
+layout: post
+title:  "Kroxylicious release 0.4.1"
+date:   2023-01-12 00:00:00 +0000
+author: "Sam Barker"
+author_url: "https://www.github.com/SamBarker"
+categories:  releases kroxylcious topic-encryption
```
Comment (robobario) [inline _posts/2024-01-12-release-0_4_1.md:7]:
```suggestion
categories:  releases kroxylicious topic-encryption
```

Comment (robobario) [reply _posts/2024-01-12-release-0_4_1.md:7]: maybe should fixup `2023-12-19-release-0_4_0.md` too

Comment (SamBarker) [reply _posts/2024-01-12-release-0_4_1.md]: As I had copy and pasted the `front matter` I've fixed this in https://github.com/kroxylicious/kroxylicious.github.io/pull/27/commits/b785f79e000fd2e3881bdd65cb1b4640f2296633

File: _posts/2024-01-12-release-0_4_1.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+### Encryption at Rest
+
+The `EnvelopeEncryption` filter works by intercepting all produce requests from applications and encrypts the Kafka records as they pass through Kroxylicious. On the consume path, the reverse happens - the filter intercepts the fetch responses and decrypts the records before they are sent to the application.
+We have more work planned to iterate on the implementation to reach production quality. Check out our [0.5.0 milestone](https://github.com/kroxylicious/kroxylicious/milestones/0.5.0) for details.
+
+The Kafka Cluster *never* sees the plain text of your records. Filter configuration is used to specify which topic(s) should be encrypted by which key.
+
+The encryption keys themselves are stored safely in a Key Management System.  0.4.0 ships with a [HashiCorp Vault](https://www.hashicorp.com/products/vault)(tm) integration.  We will provide other selected Key Management System integrations soon, however users can leverage the plug-in architecture of Kroxylicious to supply their own integrations.
```
Comment (tombentley) [inline _posts/2024-01-12-release-0_4_1.md:23]: I'm generally very wary of making statements that commit to doing _anything_, never mind implying any kind of _timeframe_. 

```suggestion
The encryption keys themselves are stored safely in a Key Management System.  0.4.0 ships with a [HashiCorp Vault](https://www.hashicorp.com/products/vault)(tm) integration.  We hope provide other selected Key Management System integrations in due course, however users can leverage the plug-in architecture of Kroxylicious to supply their own integrations.
```

File: _posts/2024-01-12-release-0_4_1.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+This release also includes some other small changes cleaning up examples and updating dependencies. 
+
+The whole Kroxylicious team is very excited to see people testing out the proxy and is really looking forward to hearing more about how & where people want to use it.
+
+### Encryption at Rest
```
Comment (robobario) [inline _posts/2024-01-12-release-0_4_1.md:16]: Should we drop this and the binary artefacts section, they are duped from 0.4.0. Maybe you could make it shorter and mention 0.4.0 plus link to the older blog if you want to maximise what this blog is delivering.

Comment (robobario) [reply _posts/2024-01-12-release-0_4_1.md:16]: ah the binaries could be a standard part of the release blog right

File: _posts/2024-01-12-release-0_4_1.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+The `EnvelopeEncryption` filter works by intercepting all produce requests from applications and encrypts the Kafka records as they pass through Kroxylicious. On the consume path, the reverse happens - the filter intercepts the fetch responses and decrypts the records before they are sent to the application.
+We have more work planned to iterate on the implementation to reach production quality. Check out our [0.5.0 milestone](https://github.com/kroxylicious/kroxylicious/milestones/0.5.0) for details.
+
+The Kafka Cluster *never* sees the plain text of your records. Filter configuration is used to specify which topic(s) should be encrypted by which key.
```
Comment (tombentley) [inline _posts/2024-01-12-release-0_4_1.md:21]: Shame we don't have documentation for this yet (https://github.com/kroxylicious/kroxylicious/issues/787), but nothing we can do here, I guess.

File: _posts/2024-01-12-release-0_4_1.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+This release also includes some other small changes cleaning up examples and updating dependencies. 
+
+The whole Kroxylicious team is very excited to see people testing out the proxy and is really looking forward to hearing more about how & where people want to use it.
+
+### Encryption at Rest
+
+The `EnvelopeEncryption` filter works by intercepting all produce requests from applications and encrypts the Kafka records as they pass through Kroxylicious. On the consume path, the reverse happens - the filter intercepts the fetch responses and decrypts the records before they are sent to the application.
```
Comment (tombentley) [inline _posts/2024-01-12-release-0_4_1.md:18]:
```suggestion
The `EnvelopeEncryption` filter works by intercepting all produce requests from applications and encrypting the Kafka records as they pass through Kroxylicious. On the consume path, the reverse happens - the filter intercepts the fetch responses and decrypts the records before they are sent to the application.
```

File: _posts/2024-01-12-release-0_4_1.md (hunk 4 above)
Comment (SamBarker) [reply _posts/2024-01-12-release-0_4_1.md:16]: I was keeping it as the core feature of 0.4.x and trying to push it in front of anyone stumbling across the announcement. 

I did re-word the binaries section to just say that they are available which felt like something the post should say explicitly, though format is entirely up for grabs.

File: _posts/2024-01-12-release-0_4_1.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+This release also includes some other small changes cleaning up examples and updating dependencies. 
+
+The whole Kroxylicious team is very excited to see people testing out the proxy and is really looking forward to hearing more about how & where people want to use it.
```
Comment (tombentley) [inline _posts/2024-01-12-release-0_4_1.md:14]:
```suggestion
The whole Kroxylicious team is very excited to see people testing out the proxy and is really looking forward to hearing more about how & where people are using it.
```

**Review by tombentley — APPROVED** [review]

File: _posts/2024-01-12-release-0_4_1.md (hunk 4 above)
Comment (robobario) [reply _posts/2024-01-12-release-0_4_1.md:16]: Yeah probably right that we should make it easy to view the demo. In that case it's a bit funky how it launches into talking about Filter. Maybe needs an intro paragraph.

File: _posts/2024-01-12-release-0_4_1.md (hunk 1 above)
Comment (SamBarker) [reply _posts/2024-01-12-release-0_4_1.md]: Fixed in https://github.com/kroxylicious/kroxylicious.github.io/pull/27/commits/8b3e26f05624bb2d7607286307ba6856e3b5b9f7

File: _posts/2024-01-12-release-0_4_1.md (hunk 4 above)
Comment (robobario) [reply _posts/2024-01-12-release-0_4_1.md:16]: like the previous blog had higher up:

The big news is this release includes our `EnvelopeEncryption` filter which provides an **Encryption-at-Rest** solution for Apache Kafka(tm) which is _transparent_ to both clients and brokers.

**Review by robobario — APPROVED** [review]

**Pull request merged**